src/main/java/org/openstreetmap/atlas/checks/validation/tag/ConstructionCheck.java

```diff
@@ -0,0 +1,234 @@
+            // 1-1-2020
+            DateTimeFormatter.ofPattern("d-M-yyyy"),
+            // 1-Jan-2020
+            DateTimeFormatter.ofPattern("d-MMM-yyyy"),
+            // 1 January 2020
+            DateTimeFormatter.ofPattern("d MMMM yyyy"));
+    private static final LocalDate TODAYS_DATE = LocalDate.now();
+    private static final List<String> DATE_TAGS = Arrays.asList("opening_date", "open_date",
+            "construction:date", "temporary:date_on", "date_on");
+    private static final List<String> CONSTRUCTION_TAGS = List.of("highway", "landuse", "building");
+
+    private final int oldConstructionDays;
+    private final int oldCheckDateMonths;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public ConstructionCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.oldConstructionDays = this.configurationValue(configuration, "oldConstructionDays",
+                OLD_CONSTRUCTION_DAYS_DEFAULT, Double::intValue);
+        this.oldCheckDateMonths = this.configurationValue(configuration, "oldCheckDateMonth",
+                OLD_CHECK_DATE_MONTHS_DEFAULT, Double::intValue);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        final Map<String, String> keySet = object.getOsmTags();
+        return !this.isFlagged(object.getOsmIdentifier()) && isConstruction(keySet);
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        this.markAsFlagged(object.getOsmIdentifier());
+
+        final Map<String, String> tags = object.getTags();
+
+        final Optional<String> dateTag = getDateTag(tags);
+        if (dateTag.isPresent())
+        {
+            final String tagDate = tags.get(dateTag.get());
+
+            final Optional<LocalDate> parsedDate = parseDate(tagDate);
+            if (parsedDate.isPresent() && parsedDate.get().isBefore(TODAYS_DATE))
+            {
+                return Optional
+                        .of(createFlag(object, this.getLocalizedInstruction(0, dateTag.get())));
```

Comment:
If the object is an Edge we will want to flag all the master edges with the same OSM ID (see [InvalidTagsCheck](https://github.com/osmlab/atlas-checks/blob/dev/src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidTagsCheck.java#L225) for an example).